### PR TITLE
fix: [CCM-7675]: Show Correct Error Message in Nodepool Recommendations

### PR DESCRIPTION
**Pull request by rajarshee-harness**
##### Jira Links:

https://harness.atlassian.net/browse/CCM-7675

##### Screenshots:

Screenshot 2022-05-19 at 1 24 55 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by rajarshee-harness** [comment]
retrigger titlecheck

**Review by jyotiarora-harness — APPROVED** [review]

**Comment by rajarshee-harness** [comment]
retrigger cypress

**Comment by rajarshee-harness** [comment]
trigger cypress

File: src/modules/75-ce/components/NodeRecommendation/NodeRecommendation.tsx [hunk 1]
```diff
@@ -209,7 +209,7 @@ const NodeRecommendationDetails: React.FC<NodeRecommendationDetailsProps> = ({
         setUpdatedState(addBufferToState(state, buffer))
         UpdatePreferenceToaster.show({ message: getString('ce.nodeRecommendation.updatePreferences'), icon: 'tick' })
       } catch (e: any) {
-        showError(getErrorInfoFromErrorObject(e))
+        showError(e?.data?.title === 'recommendation problem' ? e?.data?.detail : getErrorInfoFromErrorObject(e))
```
Comment (rishik30) [inline src/modules/75-ce/components/NodeRecommendation/NodeRecommendation.tsx:212]: let's make sure we are not dependent on some text to display the error. Might lead to unknown errors in future.

**Review by rishik30 — APPROVED** [review]

**Comment by rishik30** [comment]
Approved for now.

File: src/modules/75-ce/components/NodeRecommendation/NodeRecommendation.tsx (hunk 1 above)
Comment (rajarshee-harness) [reply src/modules/75-ce/components/NodeRecommendation/NodeRecommendation.tsx:212]: We would be adding form validation for tuning recommendations in future which would remove the need for this check and will replace it.

**Review by jyotiarora-harness — APPROVED** [review]

**Pull request merged**